From 4471aa88145d8d351c607bebf31a152db3af51e2 Mon Sep 17 00:00:00 2001
From: Xavier Hanin <xavier@apache.org>
Date: Wed, 17 Aug 2005 16:29:02 +0000
Subject: [PATCH] FIX: error retrieving dependencies with a '+' in their
 revision if the ivy file doesn't exist - fixed by maarten coene (IVY-59)

git-svn-id: https://svn.apache.org/repos/asf/incubator/ivy/trunk@484013 13f79535-47bb-0310-9956-ffa450edef68
---
 CHANGES.txt                                               | 1 +
 src/java/fr/jayasoft/ivy/resolver/RepositoryResolver.java | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/CHANGES.txt b/CHANGES.txt
index dec9bd147..ba40bbe7f 100644
--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -14,6 +14,7 @@
 - new type filtering in cachepath task
 - new cachefileset task: builds an ant fileset of artifacts in cache
 - publish task now uses srcivypattern for ivy files finding and for delivery
+- FIX: error retrieving dependencies with a '+' in their revision if the ivy file doesn't exist - fixed by maarten coene (IVY-59)
 - FIX: trying to resolve latest with no revision in pattern caused a StackOverflowError (IVY-56)
 - FIX: now handle transitive eviction (IVY-51)
 - FIX: resolve now store resolved file id in ivy variables, so that multiple resolve calls
diff --git a/src/java/fr/jayasoft/ivy/resolver/RepositoryResolver.java b/src/java/fr/jayasoft/ivy/resolver/RepositoryResolver.java
index 2191ac9b3..28d231365 100644
--- a/src/java/fr/jayasoft/ivy/resolver/RepositoryResolver.java
+++ b/src/java/fr/jayasoft/ivy/resolver/RepositoryResolver.java
@@ -78,6 +78,9 @@ public static ResolvedResource findResourceUsingPattern(String name, Repository
                 ResolvedResource found = (ResolvedResource)strategy.findLatest(rress, date); 
                 if (found == null) {
                     Message.debug("\t"+name+": no resource found for "+mrid+": pattern="+pattern);                    
+                } else if (!found.getResource().exists()) {
+                    Message.debug("\t"+name+": resource not reachable for "+mrid+": res="+res);
+                    return null; 
                 }
                 return found;
             } else {
